src/Features/Core/Completion/CompletionListContext.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    internal struct CompletionListContext
```

Comment:
This is a rather big struct.  Why not have a sealed class here?   There will only be one (as opposed to like in the codefix contexts where there might be many).